Project 14/entity/grocery-item.component.ts

```diff
@@ -0,0 +1,57 @@
+import {NameAware} from "../interface/name-aware.interface";
+import {AppElement} from "../model/app-element.enum";
+import {select} from "../helper/select";
+import {Callback} from "../model/callback.type";
+
+export class GroceryItemComponent implements NameAware {
+    private readonly boundRemoveButtonFn = this.remove.bind(this);
+    private boundLocalStorageFn: Callback<GroceryItemComponent> = () => true;
+
+    public static of(name: string): GroceryItemComponent {
+        const template = select(AppElement.GROCERY_ITEM_TEMPLATE).content.cloneNode(true) as HTMLElement;
+        return this.ofTemplate(template).setName(name);
+    }
+
+    private static ofTemplate(template: HTMLElement): GroceryItemComponent {
+        return new GroceryItemComponent(
+            select(AppElement.GROCERY_ITEM_NAME, template),
+            select(AppElement.GROCERY_ITEM_REMOVE_BUTTON, template) as HTMLButtonElement,
+            select(AppElement.GROCERY_ITEM, template)
+        );
+    }
+
+    private constructor(private name: HTMLElement,
+                        private removeButton: HTMLButtonElement,
+                        public domElement: HTMLElement) {
+        this.listen();
+    }
+
+    public remove(): void {
+        this.removeListener();
+        this.domElement.remove();
+        if (this.boundLocalStorageFn) {
+            this.boundLocalStorageFn(this);
```

Comment:
```suggestion
            this.boundLocalStorageFn?.(this);
```
и проверку можно будет убрать

Reply:
Можно даже без null-check, так как добавил default value, инициализируя переменную.